Source/WebCore/page/scrolling/ScrollAnchoringController.cpp

```diff
@@ -25,30 +25,68 @@
 
 #include "config.h"
 #include "ScrollAnchoringController.h"
+#include <wtf/text/TextStream.h>
 
 namespace WebCore {
 
-void ScrollAnchoringController::invalidateAnchorElement()
+void ScrollAnchoringController::invalidateAnchorBox()
 {
-    m_anchorElement = nullptr;
-    m_lastPositionForAnchorElement = FloatPoint();
+    m_anchorBox = nullptr;
+    m_lastPositionForAnchorBox = FloatSize();
 }
 
-void ScrollAnchoringController::selectAnchorElement()
+FloatSize ScrollAnchoringController::computeOffset(RenderBox* candidate)
 {
-    // TODO: implement
-    // perform Anchor Element Selection algorithm on dom elements within frame view's viewable region
-    // Firefox appears to only iterate through frame tree under the current frame (unclear if
-    // looking through frame tree is sufficient)
-    // unclear to me where to call this from 
+    return candidate->location() - m_frameView.rectForFixedPositionLayout().location();
+}
+
+void ScrollAnchoringController::selectAnchorBox()
+{
+    // TODO: fully implement anchor node selection algorithm in spec
+    // (specifically excluded subtrees and priority nodes)
+    auto scrollPosition = frameView().scrollPosition();
+    if (m_anchorBox || scrollPosition == IntPoint())
```

Comment:
Is this `scrollPosition == IntPoint()` in the spec? (Better as `scrollPosition.isZero()`)

You need to do some testing with RTL documents where the initial scroll position is non-zero, and the scroll origin is non-zero (and write WPTs if there aren't any).

Reply:
It is one of the suppression triggers: The scroll offset of the scrollable element being zero. https://drafts.csswg.org/css-scroll-anchoring/#suppression-triggers.

Reply:
Could you explain your second point a bit more (I'm a bit confused as to why RTL is relevant here)?